exercises/triangle/description.md

```diff
@@ -2,6 +2,16 @@ The program should raise an error if the triangle cannot exist.
 
 ## Hint
 
-The sum of the lengths of any two sides of a triangle always exceeds or 
-is equal to the length of the third side, a principle known as the _triangle
+In mathematics, a degenerate case is a limiting case in which an element of a
+class of objects is qualitatively different from the rest of the class and hence
+belongs to another, usually simpler, class.
+
+1. If the sum of the lengths of any two sides of a triangle is equal to the
+   length of the third side, there exists a degenerate triangle.
+
+2. The sum of the lengths of any two sides of a triangle always exceeds or is
```

Comment:
Good points @petertseng.  How about, "A corollary to the triangle inequality theorem is there are two classes of triangles--degenerate and nondegenerate. If the sum of the lengths of any two sides of a triangle is greater than the length of the third side (as is the case with equilateral, isosceles, and scalene triangles), that triangle is two dimensional, has area, and belongs to the nondegenerate class."

Regarding your second point, knowledge about degenerate triangles applies to the exercise because, at least for the Ruby and Coffescript exercises, the second to last test is for the degenerate case. Interestingly enough I don't see a test for the degenerate case in the Javascript exercise.   

The main reason I wanted to point out degenerate triangles do not violate the principle of triangle inequality is the tests, at least for Ruby and Coffeescript, are currently wrong.  The combination of incorrect tests and not understanding the triangle inequality confused me.  I thought either the tests were wrong or the definition of triangle inequality in the README was wrong. I incorrectly assumed the tests were right, hence the original pull request message regarding a correction to the definition of triangle inequality in the README.  After a closer look I realized a 2,2,4 triangle satisfies the degenerate condition of the triangle inequality, but In Coffeescript for example, the second to last test reads:
```coffee
it 'is illegal when violating triangle inequality 2', ->
    expect(-> new Triangle(2,2,4)).toThrow("violation of triangle inequality")
```
If it is a mistake that the Ruby and Coffeescript exercises are testing for the degenerate case, then I'm happy to make pull requests to remove those tests.  Otherwise, I'm happy to fix the tests in the languages I know if that is ok with you all. 

If you choose not to test for the degenerate case, then a way to document in the README how this knowledge about degenerate triangles applies to the exercise would be to add a "Dig Deeper" or "Extra Credit" or "Challenge" section where you explain how none of the tests checked for a degenerate triangle, but you can challenge the reader to add those tests.  You can even put the whole explanation of the corollary to the triangle inequality in the "Challenge" section itself. For OO languages you can challenge the reader to prevent degenerate triangles from being initialized directly from the Triangle class.